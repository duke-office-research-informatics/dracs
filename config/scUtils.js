import React from 'react';
import { ThemeProvider } from 'styled-components';
import { mount, shallow } from 'enzyme';
import { theme } from '../lib/dracs.es.js';

export const shallowWithTheme = (tree) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = (tree) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes // Needed so child components receive the theme prop
  });
};

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
