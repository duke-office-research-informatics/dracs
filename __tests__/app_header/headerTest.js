import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "jest-axe";
import { AppHeader, theme, IconMenu, Avatar } from "../../lib/dracs.es.js";

const imgUrl =
  "https://image.freepik.com/free-vector/soft-background-with-a-cute-blue-watercolor-stain_1048-5169.jpg";
const clickHandler = jest.fn();

describe("AppHeader", () => {
  it("matches dom snapshot", () => {
    const wrapper = mountWithTheme(<AppHeader />);
    expect(wrapper).toMatchSnapshot();
  });
  it("returns no a11y violations", async () => {
    const wrapper = mountWithTheme(<AppHeader />);
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  describe("Without Props", () => {
    it("wrapper div exists", () => {
      const wrapper = mountWithTheme(<AppHeader />);
      expect(wrapper.find("header__HeaderWrap").length).toBe(1);
    });
    it("header node exists", () => {
      const wrapper = mountWithTheme(<AppHeader />);
      expect(wrapper.find("header__Header").length).toBe(1);
    });
    it("wrapper has correct styles", () => {
      const wrapper = mountWithTheme(<AppHeader />);
      const div = wrapper.find("header__HeaderWrap");
      expect(div).toHaveStyleRule("width", "100vw");
    });
    it("header node has correct styles", () => {
      const wrapper = mountWithTheme(<AppHeader />);
      const div = wrapper.find("header__Header");
      expect(div).toHaveStyleRule("height", "56px");
      expect(div).toHaveStyleRule("width", "100%");
      expect(div).toHaveStyleRule("background-color", theme.colors.action);
      expect(div).toHaveStyleRule("background-size", "cover");
      expect(div).toHaveStyleRule("background-clip", "border-box");
      expect(div).toHaveStyleRule("background-repeat", "no-repeat");
      expect(div).toHaveStyleRule("display", "flex");
      expect(div).toHaveStyleRule("justify-content", "flex-start");
      expect(div).toHaveStyleRule("align-items", "center");
      expect(div).toHaveStyleRule("padding", "0 16px");
      expect(div).toHaveStyleRule("z-index", `${theme.zIndex.lg}`);
    });
  });

  describe("`backgroundColor` prop", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(
        <AppHeader backgroundColor={theme.colors.muted} />
      );
      expect(wrapper).toMatchSnapshot();
    });
    it("header node has correct style changes", () => {
      const wrapper = mountWithTheme(
        <AppHeader backgroundColor={theme.colors.muted} />
      );
      const div = wrapper.find("header__Header");
      expect(div).toHaveStyleRule("background-color", theme.colors.muted);
    });
  });

  describe("`backgroundImage` prop", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(<AppHeader backgroundImage={imgUrl} />);
      expect(wrapper).toMatchSnapshot();
    });
    it("returns no a11y violations", async () => {
      const wrapper = mountWithTheme(<AppHeader backgroundImage={imgUrl} />);
      const results = await axe(wrapper.html());
      expect(results).toHaveNoViolations();
    });
    it("header node has correct style changes", () => {
      const wrapper = mountWithTheme(<AppHeader backgroundImage={imgUrl} />);
      const div = wrapper.find("header__Header");
      expect(div).toHaveStyleRule("background-image", `url(${imgUrl})`);
    });
  });

  describe("`childrenLeft` prop", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(<AppHeader childrenLeft={<Avatar />} />);
      expect(wrapper).toMatchSnapshot();
    });
    it("returns no a11y violations", async () => {
      const wrapper = mountWithTheme(<AppHeader childrenLeft={<Avatar />} />);
      const results = await axe(wrapper.html());
      expect(results).toHaveNoViolations();
    });
    it("renders the child", () => {
      const wrapper = mountWithTheme(<AppHeader childrenLeft={<Avatar />} />);
      const av = wrapper.find(Avatar);
      expect(av.length).toBe(1);
    });
  });

  describe("`childrenRight` prop", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(<AppHeader childrenRight={<Avatar />} />);
      expect(wrapper).toMatchSnapshot();
    });
    it("returns no a11y violations", async () => {
      const wrapper = mountWithTheme(<AppHeader childrenRight={<Avatar />} />);
      const results = await axe(wrapper.html());
      expect(results).toHaveNoViolations();
    });
    it("renders the child", () => {
      const wrapper = mountWithTheme(<AppHeader childrenRight={<Avatar />} />);
      const av = wrapper.find(Avatar);
      expect(av.length).toBe(1);
    });
  });

  describe("`fixed` prop set to true", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(<AppHeader fixed />);
      expect(wrapper).toMatchSnapshot();
    });
    it("returns no a11y violations", async () => {
      const wrapper = mountWithTheme(<AppHeader fixed />);
      const results = await axe(wrapper.html());
      expect(results).toHaveNoViolations();
    });
    it("header node, header wrapper, fixed underlay exist", () => {
      const wrapper = mountWithTheme(<AppHeader fixed />);
      expect(wrapper.find("header__Header").length).toBe(1);
      expect(wrapper.find("header__HeaderWrap").length).toBe(1);
      expect(wrapper.find("header__FixedUnderlay").length).toBe(1);
    });
    it("header node has correct style changes", () => {
      const wrapper = mountWithTheme(<AppHeader fixed />);
      const div = wrapper.find("header__Header");
      expect(div).toHaveStyleRule("position", "absolute");
      expect(div).toHaveStyleRule("top", "0");
    });
    it("fixed underlay has correct styles", () => {
      const wrapper = mountWithTheme(<AppHeader fixed />);
      const div = wrapper.find("header__FixedUnderlay");
      expect(div).toHaveStyleRule("height", "56px");
      expect(div).toHaveStyleRule("width", "100vw");
    });
  });

  describe("`mediaQuery` string passed via prop", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(
        <AppHeader mediaQuery="@media(max-width:768px){background-color:#fff}" />
      );
      expect(wrapper).toMatchSnapshot();
    });
    it("applies the mediaQuery to the outer wrapper", () => {
      const wrapper = mountWithTheme(
        <AppHeader mediaQuery="@media(max-width:768px){background-color:#fff}" />
      );
      const div = wrapper.find("header__HeaderWrap");
      expect(div).toHaveStyleRule("background-color", "#fff", {
        media: "(max-width:768px)",
      });
    });
  });

  describe("`raised` set to true", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(<AppHeader raised />);
      expect(wrapper).toMatchSnapshot();
    });
    it("applies the correct styles to the header", () => {
      const wrapper = mountWithTheme(<AppHeader raised />);
      const div = wrapper.find("header__Header");
      expect(div).toHaveStyleRule("box-shadow", "0 0 12px 0 rgba(0,0,0,0.40)");
    });
  });

  describe("`rightIcon` props", () => {
    describe("with click handler", () => {
      it("matches dom snapshot", () => {
        const wrapper = mountWithTheme(
          <AppHeader rightIcon={<IconMenu />} onRightIconClick={clickHandler} />
        );
        expect(wrapper).toMatchSnapshot();
      });
      it("returns no a11y violations", async () => {
        const wrapper = mountWithTheme(
          <AppHeader rightIcon={<IconMenu />} onRightIconClick={clickHandler} />
        );
        const results = await axe(wrapper.html());
        expect(results).toHaveNoViolations();
      });
      it("inserts the icon into the correct wrapper with correct styling", () => {
        const wrapper = mountWithTheme(
          <AppHeader rightIcon={<IconMenu />} onRightIconClick={clickHandler} />
        );
        const icon = wrapper.find(IconMenu);
        const iconWrap = wrapper.find("header__IconRightDiv");
        const iconButton = wrapper.find("header__IconRightButton");
        expect(icon.length).toBe(1);
        expect(iconWrap.length).toBe(0);
        expect(iconButton.length).toBe(1);
        expect(iconButton).toHaveStyleRule("margin-left", "auto");
      });
      it("calls the click handler on click", () => {
        const wrapper = mountWithTheme(
          <AppHeader rightIcon={<IconMenu />} onRightIconClick={clickHandler} />
        );
        const iconButton = wrapper.find("header__IconRightButton");
        iconButton.simulate("click");
        expect(clickHandler).toBeCalled();
      });
    });

    describe("without click handler", () => {
      it("matches dom snapshot", () => {
        const wrapper = mountWithTheme(<AppHeader rightIcon={<IconMenu />} />);
        expect(wrapper).toMatchSnapshot();
      });
      it("returns no a11y violations", async () => {
        const wrapper = mountWithTheme(<AppHeader rightIcon={<IconMenu />} />);
        const results = await axe(wrapper.html());
        expect(results).toHaveNoViolations();
      });
      it("inserts the icon into the correct wrapper with correct styling", () => {
        const wrapper = mountWithTheme(<AppHeader rightIcon={<IconMenu />} />);
        const icon = wrapper.find(IconMenu);
        const iconWrap = wrapper.find("header__IconRightDiv");
        const iconButton = wrapper.find("header__IconRightButton");
        expect(icon.length).toBe(1);
        expect(iconWrap.length).toBe(1);
        expect(iconButton.length).toBe(0);
        expect(iconWrap).toHaveStyleRule("margin-left", "auto");
      });
    });
  });

  describe("`leftIcon` props", () => {
    describe("with click handler", () => {
      it("matches dom snapshot", () => {
        const wrapper = mountWithTheme(
          <AppHeader leftIcon={<IconMenu />} onLeftIconClick={clickHandler} />
        );
        expect(wrapper).toMatchSnapshot();
      });
      it("returns no a11y violations", async () => {
        const wrapper = mountWithTheme(
          <AppHeader leftIcon={<IconMenu />} onLeftIconClick={clickHandler} />
        );
        const results = await axe(wrapper.html());
        expect(results).toHaveNoViolations();
      });
      it("inserts the icon into the correct wrapper", () => {
        const wrapper = mountWithTheme(
          <AppHeader leftIcon={<IconMenu />} onLeftIconClick={clickHandler} />
        );
        const icon = wrapper.find(IconMenu);
        const iconButton = wrapper.find("header__IconLeftButton");
        expect(icon.length).toBe(1);
        expect(iconButton.length).toBe(1);
      });
      it("calls the click handler on click", () => {
        const wrapper = mountWithTheme(
          <AppHeader leftIcon={<IconMenu />} onLeftIconClick={clickHandler} />
        );
        const iconButton = wrapper.find("header__IconLeftButton");
        iconButton.simulate("click");
        expect(clickHandler).toBeCalled();
      });
    });

    describe("without click handler", () => {
      it("matches dom snapshot", () => {
        const wrapper = mountWithTheme(<AppHeader leftIcon={<IconMenu />} />);
        expect(wrapper).toMatchSnapshot();
      });
      it("returns no a11y violations", async () => {
        const wrapper = mountWithTheme(<AppHeader leftIcon={<IconMenu />} />);
        const results = await axe(wrapper.html());
        expect(results).toHaveNoViolations();
      });
      it("inserts the icon without a wrapper", () => {
        const wrapper = mountWithTheme(<AppHeader leftIcon={<IconMenu />} />);
        const icon = wrapper.find(IconMenu);
        const iconButton = wrapper.find("header__IconLeftButton");
        expect(icon.length).toBe(1);
        expect(iconButton.length).toBe(0);
      });
    });
  });

  describe("`width` set via prop", () => {
    it("matches dom snapshot", () => {
      const wrapper = mountWithTheme(<AppHeader width="200px" />);
      expect(wrapper).toMatchSnapshot();
    });
    it("applies the correct width to the correct dom node", () => {
      const wrapper = mountWithTheme(<AppHeader width="200px" />);
      const div = wrapper.find("header__Header");
      expect(div).toHaveStyleRule("width", "200px");
    });
  });
});
