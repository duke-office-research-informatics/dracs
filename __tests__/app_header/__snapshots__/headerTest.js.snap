// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader \`backgroundColor\` prop matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #696969;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  />
</div>
`;

exports[`AppHeader \`backgroundImage\` prop matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url(https://image.freepik.com/free-vector/soft-background-with-a-cute-blue-watercolor-stain_1048-5169.jpg);
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  />
</div>
`;

exports[`AppHeader \`childrenLeft\` prop matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

.c3 {
  fill: #fff;
  vertical-align: top;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  min-width: 32px;
  min-height: 32px;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: transparent;
  border-radius: 50%;
  color: transparent;
  font-size: 0;
  overflow: hidden;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  >
    <div
      className="c2"
      size={32}
    >
      <svg
        aria-label="profile icon"
        className="c3"
        color="#fff"
        height={30}
        role="img"
        viewBox="0 0 50 50"
        width={30}
      >
        <title>
          profile icon
        </title>
        <path
          d="M34.8,33.8c-1.7-0.7-3.8-1.6-3.8-2.3V27c2.5-1.9,4-4.9,4-8v-6c0-5.5-4.5-10-10-10c-5.5,0-10,4.5-10,10v6c0,3.1,1.5,6.2,4,8  v4.5c0,0.6-2.1,1.5-3.8,2.3C11.1,35.5,5,38.1,5,45v1h40v-1C45,38.1,38.9,35.5,34.8,33.8z"
        />
      </svg>
    </div>
  </header>
</div>
`;

exports[`AppHeader \`childrenRight\` prop matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c2 {
  margin-left: auto;
}

.c0 {
  width: 100vw;
}

.c4 {
  fill: #fff;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  min-width: 32px;
  min-height: 32px;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: transparent;
  border-radius: 50%;
  color: transparent;
  font-size: 0;
  overflow: hidden;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        size={32}
      >
        <svg
          aria-label="profile icon"
          className="c4"
          color="#fff"
          height={30}
          role="img"
          viewBox="0 0 50 50"
          width={30}
        >
          <title>
            profile icon
          </title>
          <path
            d="M34.8,33.8c-1.7-0.7-3.8-1.6-3.8-2.3V27c2.5-1.9,4-4.9,4-8v-6c0-5.5-4.5-10-10-10c-5.5,0-10,4.5-10,10v6c0,3.1,1.5,6.2,4,8  v4.5c0,0.6-2.1,1.5-3.8,2.3C11.1,35.5,5,38.1,5,45v1h40v-1C45,38.1,38.9,35.5,34.8,33.8z"
          />
        </svg>
      </div>
    </div>
  </header>
</div>
`;

exports[`AppHeader \`fixed\` prop set to true matches dom snapshot 1`] = `
.c2 {
  position: absolute;
  top: 0;
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

.c1 {
  height: 56px;
  width: 100vw;
}

<div
  className="c0"
>
  <div
    className="c1"
    height={56}
  />
  <header
    className="c2"
    height={56}
  />
</div>
`;

exports[`AppHeader \`leftIcon\` props with click handler matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

.c2 {
  fill: #fff;
  vertical-align: top;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  >
    <button
      className=""
      onClick={[Function]}
      tabIndex={0}
    >
      <svg
        aria-label="open customize and profile menu"
        className="c2"
        color="#fff"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          menu icon
        </title>
        <desc>
          hamburger menu icon
        </desc>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </header>
</div>
`;

exports[`AppHeader \`leftIcon\` props without click handler matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

.c2 {
  fill: #fff;
  vertical-align: top;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  >
    <svg
      aria-label="open customize and profile menu"
      className="c2"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </header>
</div>
`;

exports[`AppHeader \`mediaQuery\` string passed via prop matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

@media (max-width:768px) {
  .c0 {
    background-color: #fff;
  }
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  />
</div>
`;

exports[`AppHeader \`raised\` set to true matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  box-shadow: 0 0 12px 0 rgba(0,0,0,0.40);
  z-index: 500;
}

.c0 {
  width: 100vw;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  />
</div>
`;

exports[`AppHeader \`rightIcon\` props with click handler matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c2 {
  margin-left: auto;
}

.c0 {
  width: 100vw;
}

.c3 {
  fill: #fff;
  vertical-align: top;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  >
    <button
      className="c2"
      onClick={[Function]}
      tabIndex={0}
    >
      <svg
        aria-label="open customize and profile menu"
        className="c3"
        color="#fff"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          menu icon
        </title>
        <desc>
          hamburger menu icon
        </desc>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </header>
</div>
`;

exports[`AppHeader \`rightIcon\` props without click handler matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c2 {
  margin-left: auto;
}

.c0 {
  width: 100vw;
}

.c3 {
  fill: #fff;
  vertical-align: top;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  >
    <div
      className="c2"
    >
      <svg
        aria-label="open customize and profile menu"
        className="c3"
        color="#fff"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          menu icon
        </title>
        <desc>
          hamburger menu icon
        </desc>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </div>
  </header>
</div>
`;

exports[`AppHeader \`width\` set via prop matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 200px;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
    width="200px"
  />
</div>
`;

exports[`AppHeader matches dom snapshot 1`] = `
.c1 {
  height: 56px;
  width: 100%;
  background-color: #0680cd;
  background-image: url();
  background-size: cover;
  background-clip: border-box;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  z-index: 500;
}

.c0 {
  width: 100vw;
}

<div
  className="c0"
>
  <header
    className="c1"
    height={56}
  />
</div>
`;
